src/AggregateFunctions/AggregateFunctionMap.h

```diff
@@ -0,0 +1,207 @@
+#pragma once
+
+#include <unordered_map>
+#include <AggregateFunctions/AggregateFunctionCombinatorFactory.h>
+#include <AggregateFunctions/IAggregateFunction.h>
+#include <Columns/ColumnMap.h>
+#include <Columns/ColumnTuple.h>
+#include <Columns/ColumnVector.h>
+#include <Core/ColumnWithTypeAndName.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeMap.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Common/Arena.h>
+#include "AggregateFunctions/AggregateFunctionFactory.h"
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+template <typename KeyType>
+struct AggregateFunctionMapCombinatorData
+{
+    std::unordered_map<KeyType, AggregateDataPtr> merged_maps;
+
+    static void writeKey(KeyType key, WriteBuffer & buf) { writeBinary(key, buf); }
+    static void readKey(KeyType & key, ReadBuffer & buf) { readBinary(key, buf); }
+};
+
+template <>
+struct AggregateFunctionMapCombinatorData<String>
+{
+    std::unordered_map<String, AggregateDataPtr> merged_maps;
+
+    static void writeKey(String key, WriteBuffer & buf) { writeString(key, buf); }
+    static void readKey(String & key, ReadBuffer & buf) { readString(key, buf); }
```

Comment:
This is a bad way to serialize/deserialize strings. `readString` reads only until `\n` or `\t`:
https://github.com/ClickHouse/ClickHouse/blob/78f1ecb71f7c4e0b18e848e5c89c09cc49ed185b/src/IO/ReadHelpers.h#L471-L472

See https://github.com/ClickHouse/ClickHouse/blob/170c49db69c18bb18e13b4e55547e26c38b529c9/src/DataTypes/Serializations/SerializationString.cpp#L26
and
https://github.com/ClickHouse/ClickHouse/blob/170c49db69c18bb18e13b4e55547e26c38b529c9/src/DataTypes/Serializations/SerializationString.cpp#L34

Reply:
Good to know, how I can test the serialization?

Reply:
Serialization/deserialization is used in remote queries. For example:
```sql
SELECT sumMap(map) FROM remote('127.0.0.{1,2,3}', currentDatabase(), table_name);
```